moon-store-mac-os/moon-store-mac-os/Domain/Repositories/ProductRepository.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  ProductRepository.swift
+//  moon-store-mac-os
+//
+//  Created by Jose Luna on 12/23/24.
+//
+
+import Foundation
+
+final class ProductRepository: BaseNetworkService {
+    private let decoder: JSONDecoder
+    
+    init(decoder: JSONDecoder = .init()) {
+        self.decoder = decoder
+    }
+    
+    func getProducts() async throws -> [ProductModel] {
+        do {
+            let result = try await getData(for: MSEndpoint.products.path)
```

Comment:
Si es fetch deberia de ser fetch
```suggestion
            let data = try await getData(for: MSEndpoint.products.path)
```